nova/virt/vmwareapi/vmops.py

```diff
@@ -1027,6 +1066,17 @@ def _fetch_image_if_missing(self, context, vi):
                     ds_browser,
                     vi.cache_image_folder,
                     vi.cache_image_path.basename)
+                if (image_available and ds_util.file_size(
+                        self._session,
+                        ds_browser,
+                        vi.cache_image_folder,
+                        vi.cache_image_path.basename) ==
+                        images.INVALID_VMDK_SIZE):
+                    LOG.warning("Deleting invalid VMDK %s",
+                                vi.cache_image_path)
+                    ds_util.file_delete(self._session, vi.cache_image_path,
+                                        vi.dc_info.ref)
+                    image_available = False
```

Comment:
This is missing a try-except or a lock - I think the image-cache handling contains a lock so there are no parallel attempts to manage the files. We should probably take this lock when checking for the existence of the file already, but especially when we try to change something (i.e. delete the image file).

Reply:
Ah. looked for it, but didn't see. Thanks :+1: